accord-core/src/test/java/accord/impl/basic/Cluster.java

```diff
@@ -289,8 +278,9 @@ public static EnumMap<MessageType, Stats> run(Id[] nodes, Supplier<PendingQueue>
 
             while (!sinks.onDone.isEmpty())
             {
-                sinks.onDone.forEach(Runnable::run);
+                List<Runnable> onDone = new ArrayList<>(sinks.onDone);
```

Comment:
What issue is this addressing? We loop on processing `onDone`, so presumably we have some dependency on `onDone` for some `processPending` work?

Reply:
`new ArrayList<>(sinks.onDone);`  was to fix a `ConcurretModification` issue, the `onDone` method threw so we enqueue in `sinks.onDone` then failed.